### ISSUE TITLE
keras.LearningRateScheduler and tf.map_fn / tf.vectorized_map does not work for TPU parallelism in training time

### ISSUE THREAD

**Issue by johnypark** [johnypark/tf2-lr-schedulers#1]
Just calling them is fine, however, during training time, it does not work.

in parallel processing, the individual step, instead of vectorized step, calls the class.

tf.map_fn() / tf.vectorized_map() splits out tensors with undefined shape as a result. 

This is critical because we use matrix multiplication to generate mask tensor from compare tensor. 

A workaround is to use tf.expand_dims() before matrix multiplication and tf.squeeze() afterward. 

It works fine in training time, however, throws errors when calling the class outside the training scheme.

**Comment by johnypark** [comment]
Wrote a try-catch clause to resolve.

**Comment by johnypark** [comment]
Closing #1.
